pyof/foundation/network_types.py

```diff
@@ -276,7 +276,7 @@ def __init__(self, tlv_type=127, value=BinaryData()):
         """
         super().__init__()
         self.tlv_type = tlv_type
-        self._value = value
+        self._value = BinaryData() if value is None else value
```

Comment:
Do we really need to set `_value` as `BinaryData`?
I mean, what if we left `value` here (from the argument) and then just call the `BinaryData().pack()` if needed while packing the TLV instance?
This would go towards our goal of not using the basic types 'all around', and just use them internally.
@cemsbr what do you think?

If everybody agree, we may need to change the `pack`, `unpack` and `get_size` methods also.

Reply:
This is very well noted imho. I have not changed it because I thought this would escape this PR's scope, given the mentioned method changes. But it'd be my next change on this for sure.
I can add it up to this PR also (:

Reply:
Ok then, on the next one.